br,lightning: add docs for new s3 url params (#10572)
pingcap/docs: br/backup-and-restore-storages.md

| `external-id` | When you access data on Amazon S3 of a third-party account, you might need to specify a correct [external ID](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_create_for-user_externalid.html) to make sure you can assume [the IAM role](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles.html). `external-id` can be any random string generated by the third party that manages the S3 resources. `external-id` is optional, which means if the third party does not specify an external ID for the IAM role, you can assume the role and access the corresponding S3 data without providing this parameter. |

> **Note:**
>
> It is not recommended to pass in the access key and secret access key directly in the storage URL, because these keys are logged in plain text.

> If the access key and secret access key are not specified and no ARN for the IAM (`role-arn`) or external ID (`external-id`) is provided, the migration tools try to infer these keys from the environment in the following order:

Reviewer: ```suggestion
If neither the access key and secret access key nor IAM role ARN (`role-arn`) and external ID (`external-id`) are provided, the migration tools try to infer these keys from the environment in the following order:
```

| `--s3.sse-kms-key-id` | If `--s3.sse` is configured as `aws:kms`, this parameter is used to specify the KMS ID. |
| `--s3.acl` | The canned ACL of the upload object. For example, `private` or `authenticated-read`. |
| `--s3.provider` | The type of the S3-compatible service. The supported types are `aws`, `alibaba`, `ceph`, `netease` and `other`. |

> | `--s3.role-arn` | When you need to access data on Amazon S3 of a third-party account using [an IAM role](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles.html), you can specify the corresponding [Amazon Resource Name (ARN)](https://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html) of the IAM role with the `role-arn` parameter, such as `arn:aws:iam::888888888888:role/my-role`. For more information about using an IAM role to access data on Amazon S3 of a third-party account, see [AWS documentation](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_common-scenarios_third-party.html). |

Reviewer: ```suggestion
| `--s3.role-arn` | When you need to access Amazon S3 data from a third party using a specified [IAM role](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles.html), you can specify the corresponding [Amazon Resource Name (ARN)](https://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html) of the IAM role with the `--s3.role-arn` option, such as `arn:aws:iam::888888888888:role/my-role`. For more information about using an IAM role to access Amazon S3 data from a third party, see [AWS documentation](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_common-scenarios_third-party.html). |
```

> | `--s3.external-id` | When you access data on Amazon S3 of a third-party account, you might need to specify a correct [external ID](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_create_for-user_externalid.html) to make sure you can assume [the IAM role](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles.html). `external-id` can be any random string generated by the third party that manages the S3 resources. `external-id` is optional, which means if the third party does not specify an external ID for the IAM role, you can assume the role and access the corresponding S3 data without providing this parameter. |

Reviewer: ```suggestion
| `--s3.external-id` | When you access Amazon S3 data from a third party, you might need to specify a correct [external ID](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_create_for-user_externalid.html) to assume [the IAM role](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles.html).  In this case, you can use this `--s3.external-id` option to specify the external ID.  An external ID is an arbitrary string provided by the third party together with the IAM role ARN to access the Amazon S3 data.  Providing an external ID is optional when assuming an IAM role, which means if the third party does not specify an external ID for the IAM role, you can assume the role and access the corresponding S3 data without providing this parameter. |
```
